fix(datepicker): Fix navigation alignment with preserveWhitespaces: false

adrienverge opened the pull request:
When used with `preserveWhitespaces: false` compilation option in
Angular, the datepicker title elements are misaligned, because spaces
are missing in the justified text.

Since Angular 6, `preserveWhitespaces: false` is the default, so many
users will encounter this bug.

Let's add zero-width spaces in order to have a correct alignment in all
cases.

Closes #4443

adrienverge committed:
fix(datepicker): Fix navigation alignment with preserveWhitespaces: false

When used with `preserveWhitespaces: false` compilation option in
Angular, the datepicker title elements are misaligned, because spaces
are missing in the justified text.

Since Angular 6, `preserveWhitespaces: false` is the default, so many
users will encounter this bug.

Let's add zero-width spaces in order to have a correct alignment in all
cases.

Closes #4443

valorkin commented:
@adrienverge thanks for PR! please remote white space:true for src/demo folder and push
so QA could test this change locally

adrienverge committed:
fix(main): Reset `preserveWhitespaces` option to default

This commit reverts fee535a and 0478637, which should not be needed
anymore.

`preserveWhitespaces: false` is the default since Angular 6: let's use
the default!

adrienverge commented:
@valorkin Sure.

I've added a commit:

---

### fix(main): Reset `preserveWhitespaces` option to default

This commit reverts fee535a and 0478637, which should not be needed
anymore.

`preserveWhitespaces: true` is the default since Angular 6: let's use
the default!
